main.js

```diff
@@ -149,14 +143,55 @@ const endWsCnx = () => {
     );
 }
 
-const generate = () => {
+const generate = async () => {
+    document.getElementById("generate").setAttribute('disabled', 'disabled');
     endWsCnx();
 
+    // Await server closing the WS
+    for (let i = 0; i < 50; i++) {
+        if (websocket.readyState === websocket.OPEN) {
+            await sleep(100);
+        } else {
+            break;
+        }
+    }
+
     audioContext?.close();
     pcmWorker?.port.close();
     pcmWorker?.disconnect();
     mediaSource?.mediaStream.getTracks().forEach((track) => track.stop());
     mediaSource?.disconnect();
 
-    document.getElementById("generate").setAttribute('disabled', 'disabled');
+    await callDigest();
 }
+
+const callDigest = async () => {
+    const response = await fetch('https://api.nabla.com/v1/copilot-api/server/digest', {
```

Comment:
Here you don't call the async endpoint, do you? 
It's probably ok to keep it like that for now, but we could also add an example of integration of the async endpoint.

Reply:
we can do that but it would require setting up a proxy to receive the async callback